interface/frontend/src/components/open-image/OpenDICOM.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <div class="DICOM-container">
+    <div class="DICOM" ref="DICOM"></div>
+  </div>
+</template>
+
+<script>
+  import { EventBus } from '../../main.js'
+  const cornerstone = require('cornerstone-core')
+  const Q = require('q')
+
+  export default {
+    name: 'open-dicom',
+    data () {
+      return {
+        dicom: {
```

Comment:
I just went through cornerstone's example and have a better understanding of the library. Since we are using vuejs, and since we want this component to be re-usable, it is crucial that we might want to construct this entire object upon rendering the image instead of having a mocked version of it here with some binding to other places to get data.

My suggested implementation looks something like this:

+ This vue comp will accept parameters from parent component, which is the path to the image
+ With that, we move `dicom` to computed and simply do all the fetch and calculation there. With this we might not need to `watch` at all
+ On the same line of thought, we can simply construct the dicom object everytime the path changed, and then apply it using cornerstone.

The above should reduce the intertwine between static data and computed data that we are having here.

Reply:
Refer to this for computed variable in vue: https://vuejs.org/v2/guide/computed.html

Reply:
I've designed data flow through `watch` in order to ensure that data has been uploaded by asynchronous `axios.get`, now for the same goal, I've rewritten it in a promise-based style (with a chain of promises). It also turned out that each click a new canvas instance was created, this unexpected behaviour is fixed now.